lib/webdriverio.js

```diff
@@ -207,19 +207,21 @@ let WebdriverIO = function (args, modifier) {
         let timestamp = new Date().toJSON().replace(/:/g, '-')
         let fileName = `ERROR_${capId}_${timestamp}.png`
         let filePath = path.join(screenshotPath, fileName)
+        let screenshotData = err.screenshot
 
         /**
          * don't take a new screenshot if we already got one from Selenium
          */
-        if (typeof err.screenshot === 'string') {
+        if (typeof screenshotData === 'string') {
             let screenshot = new Buffer(err.screenshot, 'base64')
             fs.writeFileSync(filePath, screenshot)
         } else {
             client.next(prototype.saveScreenshot, [filePath], 'saveScreenshot')
+            screenshotData = fs.readFileSync(filePath, 'utf8')
         }
         this.logger.log(`\tSaved screenshot: ${fileName}`)
         this.emit('screenshot', {
```

Comment:
There is no need to emit the `screenshot` event when the client manually takes one (like in line 220)